### PR TITLE
🏃 Update test scripts

### PR DESCRIPTION
**Pull request by maelk**
**What this PR does / why we need it**:
Adds golint.sh and codegen.sh and make gofmt.sh fail if files
are changed

### CODE REVIEW

File: hack/codegen.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+set -eux
+
+IS_CONTAINER=${IS_CONTAINER:-false}
+ARTIFACTS=${ARTIFACTS:-/tmp}
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+
+if [ "${IS_CONTAINER}" != "false" ]; then
+  export XDG_CACHE_HOME=/tmp/.cache
+  eval "$(go env)"
+  cd "${GOPATH}"/src/github.com/metal3-io/cluster-api-provider-metal3
```
Comment (fmuyassarov) [inline hack/codegen.sh:12]: Don't you need to export `GOPATH`?

File: hack/codegen.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  INPUT_FILES="\
+  config/certmanager/*.yaml
+  config/crd/*.yaml
+  config/crd/bases/*.yaml
+  config/crd/patches/*.yaml
+  config/default/*.yaml
+  config/manager/*.yaml
+  config/rbac/*.yaml
+  config/webhook/*.yaml
+  config/*.yaml
```
Comment (fmuyassarov) [inline hack/codegen.sh:22]: I don't see `/bmo`. I'm not sure if it also needs to be added or not?

File: hack/codegen.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+set -eux
+
+IS_CONTAINER=${IS_CONTAINER:-false}
+ARTIFACTS=${ARTIFACTS:-/tmp}
```
Comment (fmuyassarov) [inline hack/codegen.sh:6]: Double quotes around curly brackets?

File: hack/golint.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+set -eux
+
+IS_CONTAINER=${IS_CONTAINER:-false}
```
Comment (fmuyassarov) [inline hack/golint.sh:5]: Double quotes around curly brackets?

File: hack/codegen.sh (hunk 1 above)
Comment (maelk) [reply hack/codegen.sh:12]: it's defined as part of line 11

File: hack/codegen.sh (hunk 2 above)
Comment (maelk) [reply hack/codegen.sh:22]: this is v1a3, there is no BMO, and no ipam

File: hack/codegen.sh (hunk 3 above)
Comment (maelk) [reply hack/codegen.sh:6]: I can add that indeed

File: hack/golint.sh (hunk 4 above)
Comment (maelk) [reply hack/golint.sh:5]: ok

File: hack/codegen.sh (hunk 3 above)
Comment (furkatgofurov7) [reply hack/codegen.sh:6]: Do we need double quotes here? Check [this](https://github.com/metal3-io/cluster-api-provider-metal3/blob/036ea1866c391e8d22213cfc7c0ebba597292b1d/hack/codegen.sh#L5)

File: hack/golint.sh (hunk 4 above)
Comment (furkatgofurov7) [reply hack/golint.sh:5]: Check [this](https://github.com/metal3-io/cluster-api-provider-metal3/blob/036ea1866c391e8d22213cfc7c0ebba597292b1d/hack/golint.sh#L5) as well

**Pull request merged**